### DISCUSSION TITLE
Stream output from playing XBoard engine?

### DISCUSSION THREAD

**Discussion by laniakea64** [niklasf/python-chess#899] [Q&A]
Really appreciate this awesome library!  It seems to have functionality for basically everything chess-related I can think of.  Nice!

I would like to use python-chess to facilitate playing various chess engines against each other while streaming/capturing the engine outputs.  Currently I am using `engine.analysis()` per https://github.com/niklasf/python-chess/issues/405 .  It works well with UCI engines.  But I would also like to add Crafty to the mix, but clock limit with `engine.analysis()` is not supported for XBoard engines -
https://github.com/niklasf/python-chess/blob/3c902fde7a4744b285d933eb6116ec78a87871eb/chess/engine.py#L2254-L2255
And the code for XBoard engine analysis looks very different from the code for XBoard engine playing.

So how to stream info from a XBoard engine while it's playing?

Thanks!

**Comment by niklasf** [comment]
Hi. Mhh ... unfortunately I don't think it's currently possible. Of course that's an issue with the `chess.engine` abstraction, not the XBoard protocol itself, but I am not quite sure what's the best way to go about it.
